libs/libc/string/lib_memchr.c

```diff
@@ -49,17 +49,14 @@ FAR void *memchr(FAR const void *s, int c, size_t n)
 {
   FAR const unsigned char *p = (FAR const unsigned char *)s;
 
-  if (s)
+  while (n--)
     {
-      while (n--)
+      if (*p == (unsigned char)c)
```

Comment:
not related to this PR, but just curious why we are using `unsigned char` and not `char` here

Reply:
the code may want to avoid depend on whether char is extend as unsigned or signed.